components/ingest-service/server/chef.go

```diff
@@ -341,89 +340,154 @@ func (s *ChefIngestServer) StartReindex(ctx context.Context, req *ingest.StartRe
 		}
 	}
 
-	var isFailed bool
-	// start the reindex process for each and every index available in the DB and which needs reindexing
-	// loop through the indices
-	for _, index := range indexList {
-		// 1. Check if the index needs reindexing
-		isEligible, _, _ := s.VersionComparision(index)
-		if !isEligible {
-			log.WithFields(log.Fields{"index": index}).Info("Index does not need reindexing")
-			continue
-		}
+	backgroundCtx, cancel := context.WithTimeout(context.Background(), 24*time.Hour)
+
+	errChan := make(chan error, 1)
 
-		// 2. Get the aliases and update the database
-		alias, err := s.getAliases(ctx, index, requestID)
+	// Start the reindexing process in a background goroutine
+	go func() {
+		defer cancel() // Ensure context is canceled when goroutine completes
+
+		// Run the reindexing process with the detached context
+		err := s.runReindexingProcess(backgroundCtx, indexList, requestID, errChan)
 		if err != nil {
-			isFailed = true
-			log.WithFields(log.Fields{"index": index}).WithError(err).Error("Failed to fetch aliases for index: ", index)
-			continue
+			log.WithFields(log.Fields{"requestId": requestID}).WithError(err).Error("Reindexing process failed")
+			errChan <- err
 		}
-		tempIndex := index + "_temp"
+	}()
 
-		// Step 3: Create Temporary Index
-		if err := s.createIndex(ctx, tempIndex, index, requestID, SRC_TO_TEMP, index); err != nil {
-			isFailed = true
-			continue
-		}
+	return &ingest.StartReindexResponse{
+		Message: "Reindexing started successfully",
+	}, nil
+}
 
-		// Step 4: Reindex from Source to Temporary Index
-		if err := s.processReindexing(ctx, tempIndex, index, requestID, REINDEX_SRC_TEMP, index); err != nil {
-			isFailed = true
-			log.WithFields(log.Fields{"index": index}).WithError(err).Error("Failed to reindex from source to temporary index: ", index)
-			continue
-		}
+func (s *ChefIngestServer) runReindexingProcess(ctx context.Context, indexList []string, requestID int, errChan chan error) error {
+	// Start a goroutine for heartbeat updates
+	heartbeatDone := make(chan struct{})
+	defer close(heartbeatDone)
+
+	// Create a ticker for heartbeat updates - every 2 minutes
+	go func() {
+		ticker := time.NewTicker(2 * time.Minute)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ctx.Done():
+				log.WithFields(log.Fields{"requestId": requestID}).Info("Heartbeat stopped due to context cancellation")
+				return
+			case <-heartbeatDone:
+				log.WithFields(log.Fields{"requestId": requestID}).Info("Heartbeat stopped as process completed")
+				return
+			case <-ticker.C:
+				// Update heartbeat for all indices in this request
+				for _, index := range indexList {
+					if err := s.db.UpdateReindexStatus(requestID, index, "heartbeat", time.Now()); err != nil {
```

Comment:
We don't need this UpdateReindexStatus, even if we need heartbeat to keep the routine going. Update is not required.